slice/internal/controller/workload_controller.go

```diff
@@ -64,34 +85,74 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	log.V(2).Info("Reconcile Workload")
 
 	if r.shouldFinalize(wl) {
-		if controllerutil.ContainsFinalizer(wl, CleanupSliceFinalizerName) {
+		if controllerutil.ContainsFinalizer(wl, SliceControllerName) {
 			err = r.client.Delete(ctx, r.newEmptySlice(wl))
 			if client.IgnoreNotFound(err) != nil {
 				return ctrl.Result{}, err
 			}
-			controllerutil.RemoveFinalizer(wl, CleanupSliceFinalizerName)
+			controllerutil.RemoveFinalizer(wl, SliceControllerName)
 			if err := r.client.Update(ctx, wl); err != nil {
-				return ctrl.Result{}, err
+				if !apierrors.IsNotFound(err) {
+					log.Error(err, "Failed to remove finalizer")
+				}
+				return ctrl.Result{}, client.IgnoreNotFound(err)
 			}
 			log.V(5).Info("Removed finalizer")
 		}
 		return ctrl.Result{}, nil
 	}
 
-	if controllerutil.AddFinalizer(wl, CleanupSliceFinalizerName) {
-		if err := r.client.Update(ctx, wl); err != nil {
-			log.V(5).Info("Added finalizer")
-			return ctrl.Result{}, err
+	ac, err := r.sliceAC(ctx, wl)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	if ac == nil {
+		log.V(5).Info("Admission check not found – ignoring reconciliation for now")
+		return reconcile.Result{}, nil
+	}
+
+	log = log.WithValues("admissionCheck", ac.Name)
+	ctrl.LoggerInto(ctx, log)
+
+	if ac.State == kueue.CheckStateReady {
+		log.V(5).Info("Admission check is ready — nothing to do")
+		return reconcile.Result{}, nil
+	}
+
+	if controllerutil.AddFinalizer(wl, SliceControllerName) {
+		if err = r.client.Update(ctx, wl); err != nil {
+			if !apierrors.IsNotFound(err) {
+				log.Error(err, "Failed to add finalizer")
+			}
+			return ctrl.Result{}, client.IgnoreNotFound(err)
 		}
+		log.V(5).Info("Added finalizer")
+		return ctrl.Result{}, nil
 	}
 
-	return ctrl.Result{}, r.createSliceIfNotExist(ctx, wl)
+	err = r.createSliceIfNotExist(ctx, wl, ac)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, client.IgnoreNotFound(r.handleSuccess(ctx, wl, ac))
```

Comment:
Creating slice isn't enough for success - the slice needs to transition to state Ready.

Reply:
Done